### ISSUE TITLE
Add vpool api should check if the mountpoint is empty

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework#1063]
### Problem

In reference to https://github.com/openvstorage/framework/issues/626, I found out that the api call for add vpool does not validate the contents or the existence of a directory in under /mnt (the check_mtpt call does though)
### Possible solution

Call check_mtpt during add_vpool to post the correct feedback to the user as to why the vpool creation failed.

**Comment by khenderick** [comment]
So, to summarize:
- both wizard AND add_vpool pre-validation logic should use check_mtpt.
- check_mtpt should validate whether the directory exists or not
- We want the directory to be non-existing

**Comment by khenderick** [comment]
Fixed by #1071, packaged in openvstorage-2.7.4-rev.4185.1ded339

**Comment by JeffreyDevloo** [comment]
### Steps
- Setup new cluster
- Added disk roles
- Added backend
- Added vpool
- Created new directory 'test-check' in /mnt 
- Tried adding new vpool via webapi

Code to test:

```
from ci.helpers.api import OVSClient
from ovs.dal.lists.storagerouterlist import StorageRouterList
from ovs.dal.lists.albabackendlist import AlbaBackendList
# Loading integration test ovsclient for password login

# Requires a backend to be already setup

class StoragerouterHelper(object):

    @staticmethod
    def get_storagerouter_by_ip(storagerouter_ip):
        """

        :param storagerouter_ip: ip of a storagerouter
        :type storagerouter_ip: str
        :return: storagerouter object
        :rtype: ovs.dal.hybrids.storagerouter.StorageRouter
        """
        return StorageRouterList.get_by_ip(storagerouter_ip)


class BackendHelper(object):

    @staticmethod
    def get_albabackend_by_name(albabackend_name):
        """
        Get a Albabackend by name

        :param albabackend_name: albabackend name
        :type albabackend_name: str
        :return: alba backend object
        :rtype: ovs.dal.hybrids.albabackend
        """

        try:
            return [alba_backend for alba_backend in AlbaBackendList.get_albabackends()
                    if alba_backend.name == albabackend_name][0]
        except IndexError:
            error_msg = "No Alba backend found with name: {0}".format(albabackend_name)
            raise NameError(error_msg)

class VPoolTest(object):

    def __init__(self):
        self.vpool_details = {
            "backend_name": "mybackend",
            "preset": "testpreset",
            "storage_ip": "10.100.199.151",
            "fragment_cache": {
                "strategy": {"cache_on_read": True, "cache_on_write": True},
                "location": "disk"
            },
            "storagedriver": {
                "sco_size": 4,
                "cluster_size": 4,
                "volume_write_buffer": 512,
                "strategy": "none",
                "global_write_buffer": 2,
                "global_read_buffer": 0,
                "deduplication": "non_dedupe",
                "dtl_transport": "tcp",
                "dtl_mode": "a_sync"
            }
        }
        self.storagerouter_ip = '10.100.199.151'
        self.vpool_name = 'test-check'
        self.api = OVSClient(self.storagerouter_ip, 'admin', 'admin')

    def validate(self, old=True):
        if old is True:
            self.test_mountpoint_old_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)
        else:
            self.test_mountpoint_new_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)

    @staticmethod
    def test_mountpoint_old_flow(storagerouter_ip, vpool_name, vpool_details, api):
        """
        Will add a vpool without checking whether or not the mount point is in use
        :return:
        """

        # Add vpool because mointpoint is not in use
        # Build ADD_VPOOL parameters
        call_parameters = {
            "call_parameters": {
                "vpool_name": vpool_name,
                "type": "alba",
                "backend_connection_info": {
                    "host": "",
                    "port": 80,
                    "username": "",
                    "password": "",
                    "backend": {
                        "backend": BackendHelper.get_albabackend_by_name(vpool_details['backend_name']).guid,
                        "metadata": vpool_details['preset']
                    }
                },
                "storage_ip": vpool_details['storage_ip'],
                "storagerouter_ip": storagerouter_ip,
                "readcache_size": int(vpool_details['storagedriver']['global_read_buffer']),
                "writecache_size": int(vpool_details['storagedriver']['global_write_buffer']),
                "fragment_cache_on_read": vpool_details['fragment_cache']['strategy']['cache_on_read'],
                "fragment_cache_on_write": vpool_details['fragment_cache']['strategy']['cache_on_write'],
                "config_params": {
                    "dtl_mode": vpool_details['storagedriver']['dtl_mode'],
                    "sco_size": int(vpool_details['storagedriver']['sco_size']),
                    "dedupe_mode": vpool_details['storagedriver']['deduplication'],
                    "cluster_size": int(vpool_details['storagedriver']['cluster_size']),
                    "write_buffer": int(vpool_details['storagedriver']['volume_write_buffer']),
                    "dtl_transport": vpool_details['storagedriver']['dtl_transport'],
                    "cache_strategy": vpool_details['storagedriver']['strategy']
                }
            }
        }

        task_guid = api.post(
            api='/storagerouters/{0}/add_vpool/'.format(
                StoragerouterHelper.get_storagerouter_by_ip(storagerouter_ip).guid),
            data=call_parameters
        )
        task_result = api.wait_for_task(task_id=task_guid, timeout=120)
        if not task_result[0]:
            error_msg = "vPool `{0}` has failed to create on storagerouter `{1}`. Call returned {2}".format(vpool_name,storagerouter_ip, task_result)
            raise RuntimeError(error_msg)
        else:
            print "Creation of vPool `{0}` should have succeeded on storagerouter `{1}`".format(vpool_name,
                                                                                                storagerouter_ip)
            return storagerouter_ip, "/mnt/{0}".format(vpool_name)

    @staticmethod
    def test_mountpoint_new_flow(storagerouter_ip, vpool_name, vpool_details, api):
        """
        Will add a vpool with checking whether or not the mount point is in use
        :return:
        """
        # Check for existing mountpoint
        data = {
            'name': vpool_name
        }
        task_id = api.post(api='/storagerouters/{0}/check_mtpt/'.format(StoragerouterHelper.get_storagerouter_by_ip(storagerouter_ip).guid), data=data)
        task_result = api.wait_for_task(task_id)
        # Both are true, means that the mountpoint is not in use
        if task_result[0] is True:
            if task_result[1] is True:
                # Add vpool because mointpoint is not in use
                # Build ADD_VPOOL parameters
                call_parameters = {
                    "call_parameters": {
                        "vpool_name": vpool_name,
                        "type": "alba",
                        "backend_connection_info": {
                            "host": "",
                            "port": 80,
                            "username": "",
                            "password": "",
                            "backend": {
                                "backend": BackendHelper.get_albabackend_by_name(vpool_details['backend_name']).guid,
                                "metadata": vpool_details['preset']
                            }
                        },
                        "storage_ip": vpool_details['storage_ip'],
                        "storagerouter_ip": storagerouter_ip,
                        "readcache_size": int(vpool_details['storagedriver']['global_read_buffer']),
                        "writecache_size": int(vpool_details['storagedriver']['global_write_buffer']),
                        "fragment_cache_on_read": vpool_details['fragment_cache']['strategy']['cache_on_read'],
                        "fragment_cache_on_write": vpool_details['fragment_cache']['strategy']['cache_on_write'],
                        "config_params": {
                            "dtl_mode": vpool_details['storagedriver']['dtl_mode'],
                            "sco_size": int(vpool_details['storagedriver']['sco_size']),
                            "dedupe_mode": vpool_details['storagedriver']['deduplication'],
                            "cluster_size": int(vpool_details['storagedriver']['cluster_size']),
                            "write_buffer": int(vpool_details['storagedriver']['volume_write_buffer']),
                            "dtl_transport": vpool_details['storagedriver']['dtl_transport'],
                            "cache_strategy": vpool_details['storagedriver']['strategy']
                        }
                    }
                }

                task_guid = api.post(
                    api='/storagerouters/{0}/add_vpool/'.format(StoragerouterHelper.get_storagerouter_by_ip(storagerouter_ip).guid),
                    data=call_parameters
                )
                task_result = api.wait_for_task(task_id=task_guid, timeout=120)
                if not task_result[0]:
                    error_msg = "vPool `{0}` has failed to create on storagerouter `{1}`".format(vpool_name, storagerouter_ip)
                    raise RuntimeError(error_msg)
                else:
                    print "Creation of vPool `{0}` should have succeeded on storagerouter `{1}`".format(vpool_name, storagerouter_ip)
                    return storagerouter_ip, "/mnt/{0}".format(vpool_name)
            else:
                # Mountpoint already in use
                raise AssertionError('Mountpoint {0} already in use'.format(vpool_name))
```

### Output

```
In [5]: VPoolTest().validate(old=True)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-e01f4b1a8522> in <module>()
----> 1 VPoolTest().validate(old=True)

<ipython-input-4-7edb74999ff3> in validate(self, old)
     69     def validate(self, old=True):
     70         if old is True:
---> 71             self.test_mountpoint_old_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)
     72         else:
     73             self.test_mountpoint_new_flow(self.storagerouter_ip, self.vpool_name, self.vpool_details, self.api)

<ipython-input-4-7edb74999ff3> in test_mountpoint_old_flow(storagerouter_ip, vpool_name, vpool_details, api)
    122         if not task_result[0]:
    123             error_msg = "vPool `{0}` has failed to create on storagerouter `{1}`. Call returned {2}".format(vpool_name,storagerouter_ip, task_result)
--> 124             raise RuntimeError(error_msg)
    125         else:
    126             print "Creation of vPool `{0}` should have succeeded on storagerouter `{1}`".format(vpool_name,

RuntimeError: vPool `test-check` has failed to create on storagerouter `10.100.199.151`. Call returned (False, u'The mountpoint for vPool test-check already exists')
```

### Test result

Test passed.
